frame/vesting/src/lib.rs

```diff
@@ -326,45 +395,91 @@ impl<T: Config> VestedTransfer for Pallet<T> {
 			to: to.clone(),
 			asset,
 			schedule,
+			vesting_schedule_id,
 		});
 
 		Ok(())
 	}
 }
 
 impl<T: Config> Pallet<T> {
-	fn do_claim(who: &AccountIdOf<T>, asset: AssetIdOf<T>) -> Result<BalanceOf<T>, DispatchError> {
-		let locked = Self::locked_balance(who, asset);
+	fn do_claim(
+		who: &AccountIdOf<T>,
+		asset: AssetIdOf<T>,
+		vesting_schedule_id: Option<T::VestingScheduleId>,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		let locked = Self::locked_balance(who, asset, vesting_schedule_id);
+
 		if locked.is_zero() {
 			// cleanup the storage and unlock the fund
 			<VestingSchedules<T>>::remove(who, asset);
 			T::Currency::remove_lock(VESTING_LOCK_ID, asset, who)?;
 		} else {
 			T::Currency::set_lock(VESTING_LOCK_ID, asset, who, locked)?;
 		}
+
 		Ok(locked)
 	}
 
-	/// Returns locked balance based on current block number.
-	fn locked_balance(who: &AccountIdOf<T>, asset: AssetIdOf<T>) -> BalanceOf<T> {
+	/// Returns total locked balance and balance per vesting schedule, based on current block number
+	fn locked_balance(
+		who: &AccountIdOf<T>,
+		asset: AssetIdOf<T>,
+		vesting_schedule_id: Option<T::VestingScheduleId>,
```

Comment:
Consider a more strongly-typed enum here:

```rust
enum Schedules<Id> {
    All,
    One(Id),
}
```

Reply:
Could also consider a `BoundedVec<_>` to allow for claiming any amount at once, rather than just all or nothing